Source/ACE.Server/WorldObjects/SkillAlterationDevice.cs

```diff
@@ -0,0 +1,250 @@
+            SkillToBeAltered = (Skill)(GetProperty(PropertyInt.SkillToBeAltered) ?? 0);
+        }
+
+        /// <summary>
+        /// This is raised by Player.HandleActionUseItem.<para />
+        /// The item should be in the players possession.
+        /// </summary>
+        public override void UseItem(Player player)
+        {
+            var currentSkill = player.GetCreatureSkill(SkillToBeAltered);
+
+            //Check to make sure we got a valid skill back
+            if (currentSkill == null)
+            {
+                player.Session.Network.EnqueueSend(new GameEventUseDone(player.Session, WeenieError.YouFailToAlterSkill));
+                return;
+            }
+
+            //Gather costs associated with manipulating currently selected skill
+            var currentSkillCost = currentSkill.Skill.GetCost();
+
+            switch (TypeOfAlteration)
+            {
+                case SkillAlterationType.Specialize:
+                    //Check to make sure player won't exceed limit of 70 specialized credits after operation
+                    if (currentSkillCost.SpecializationCost + GetTotalSpecializedCredits(player) > 70)
+                    {
+                        player.Session.Network.EnqueueSend(new GameEventWeenieErrorWithString(player.Session, WeenieErrorWithString.TooManyCreditsInSpecializedSkills, currentSkill.Skill.ToSentence()));
+                        player.Session.Network.EnqueueSend(new GameEventUseDone(player.Session, WeenieError.YouFailToAlterSkill));
+                        break;
+                    }
+
+                    //Check to see if the skill is ripe for specializing
+                    if (currentSkill.AdvancementClass == SkillAdvancementClass.Trained)
+                    {
+                        if (player.AvailableSkillCredits >= currentSkillCost.SpecializationCost)
+                        {
+                            if (player.SpecializeSkill(currentSkill.Skill, currentSkillCost.SpecializationCost))
+                            {
+                                //Specialization was successful, notify the client
+                                player.Session.Network.EnqueueSend(new GameMessagePrivateUpdateSkill(player, currentSkill.Skill, SkillAdvancementClass.Specialized, 0, 0, 0));
+                                player.Session.Network.EnqueueSend(new GameMessagePrivateUpdatePropertyInt(player, PropertyInt.AvailableSkillCredits, player.AvailableSkillCredits ?? 0));
+                                player.Session.Network.EnqueueSend(new GameEventWeenieErrorWithString(player.Session, WeenieErrorWithString.YouHaveSucceededSpecializing_Skill, currentSkill.Skill.ToSentence()));
+                                player.Session.Network.EnqueueSend(new GameEventUseDone(player.Session, WeenieError.None));
+
+                                //Destroy the gem we used successfully
+                                Destroy();
```

Comment:
Should be using player.TryRemoveItemFromInventoryWithNetworking(this, 1) without the line following the Destroy()

Reply:
Right you are. I knew about that method, but spaced on using it. Thanks for the guidance. I've fixed all three references and pushed a new commit.